### PR TITLE
Add skipped test for CA1825 firing on compiler generated code

### PR DESCRIPTION
**Pull request by mavasani**
See https://github.com/dotnet/roslyn/issues/10214 for context.

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @dotnet/roslyn-analysis for review.

File: src/System.Runtime.Analyzers/UnitTests/AvoidZeroLengthArrayAllocationsTests.cs [hunk 1]
```diff
@@ -217,5 +218,32 @@ void M1()
...
+    End Class
+End Namespace
+";
+            const string source = @"
+Class C
+    Sub M1()
+        Dim a = ""STR "".TrimEnd()               ' Compiler seems to generate a param array with size 0 for the TrimEnd invocation.
+    End Sub
+End Class";
+
```
Comment (lgolding) [inline src/System.Runtime.Analyzers/UnitTests/AvoidZeroLengthArrayAllocationsTests.cs]: Would a more straightforward test be this?
```
class C
{
    private void F(params string[] args) { }

    private void G()
    {
        F();
    }
}
```
By more straightforward, I mean that it explicitly shows the creation of a `params` array of length 0.

File: src/System.Runtime.Analyzers/UnitTests/AvoidZeroLengthArrayAllocationsTests.cs [hunk 2]
```diff
@@ -217,5 +218,32 @@ void M1()
...
+            string arrayEmptySource = IsArrayEmptyDefined() ? string.Empty : arrayEmptySourceRaw;
+
```
Comment (lgolding) [inline src/System.Runtime.Analyzers/UnitTests/AvoidZeroLengthArrayAllocationsTests.cs]: Ok, I'm confused. Are you saying that in your sample code, the compiler would generate an empty array even on platforms where `Array.Empty` does not exist?

Comment (mavasani) [inline src/System.Runtime.Analyzers/UnitTests/AvoidZeroLengthArrayAllocationsTests.cs]: `Array.Empty` exists only on .NET 4.6 and later. So the tests will fail if run against an older version of the framework. The code here ensures that we always have a single definition of `Array.Empty` in the compilation, regardless of the target framework.

**Comment by mavasani** [comment]
We decided to apply a workaround till the IOperation bug is fixed. I'll send a separate PR for the workaround + test, closing this one.

**Pull request closed without merging**